classes/robot/crawler.php

```diff
@@ -634,6 +636,23 @@ public function scrape($url) {
         return $result;
     }
 
+    /**
+     * Checks whether robot should authenticate or not.
+     * Bot should authenticate if URL it is crawling over is local URL
+     * And bot should not authenticate when crawling over external URLs.
+     *
+     * @param string $url
+     * @return boolean
+     */
+    public function should_be_authenticated($url) {
+        global $CFG;
+        $mydomain = parse_url($CFG->wwwroot, PHP_URL_HOST);
+        $urldomain = parse_url($url, PHP_URL_HOST);
+        if ( $mydomain == $urldomain ) {
```

Comment:
I'm pretty sure this implementation will fail in some cases, eg if there was a wwwroot = 'http://moodle.com/sub1' and you had a url of ''http://moodle.com/sub2' this would return true when it should return false. Add this as another test case to the provider and see, and then if it fails try the implementation we came up with before: if(strpos($url, $CFG->wwwroot.'/') === 0 || $url === $CFG->wwwroot) {